omod/src/test/java/org/openmrs/module/idgen/rest/search/LogEntrySearchHandlerTest.java

```diff
@@ -82,7 +82,78 @@ public void getSearchConfig_shouldReturnLogEntryByDateGenerated() throws Excepti
 		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
 		req.addParameter("generatedBy", USER_UUID);
+		req.addParameter("comment", "New");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(2, Util.getResultsSize(result));
+	}
+
+	@Test
+	public void getSearchConfig_shouldReturnLogEntryByGeneratedByAndComment() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("generatedBy", USER_UUID);
+		req.addParameter("comment", "New");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(2, Util.getResultsSize(result));
+	}
+	@Test
+	public void getSearchConfig_shouldReturnLogEntryByGeneratedByAndCommentAndIdentifier() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("generatedBy", USER_UUID);
+		req.addParameter("comment", "New");
+		req.addParameter("identifier", "100");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(2, Util.getResultsSize(result));
+	}
+
+	@Test
+	public void getSearchConfig_shouldReturnLogEntryBySourceAndComment() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("source", LOG_ENTRY_SOURCE_UUID);
+		req.addParameter("comment", "New");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(result);
+		Assert.assertEquals(2, Util.getResultsSize(result));
+	}
+
+
+	@Test
+	public void getSearchConfig_shouldReturnLogEntryBySourceAndCommentAndIdentifier() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("source", LOG_ENTRY_SOURCE_UUID);
+		req.addParameter("comment", "New");
+		req.addParameter("identifier", "100");
 		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(result);
 		Assert.assertEquals(2, Util.getResultsSize(result));
 	}
+
+	@Test
+	public void getSearchConfig_shouldReturnNoLogEntriesForInvalidGeneratedBy() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("generatedBy", "invalid_user");
+		req.addParameter("comment", "New");
+		req.addParameter("identifier", "100");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(0, Util.getResultsSize(result));
+	}
+	@Test
+	public void getSearchConfig_shouldReturnNoLogEntriesForInvalidSource() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("source", "invalid_source");
+		req.addParameter("comment", "New");
+		req.addParameter("identifier", "100");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertEquals(0, Util.getResultsSize(result));
+	}
+
+	@Test
+	public void getSearchConfig_shouldReturnNoLogEntriesIfDateRangeIsWrong() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("fromDate", "2017-10-21T12:00:00.433");
```

Comment:
I mean invalid dates. As in strings which cannot be parsed into dates.

Reply:
The above test if for date range ie if no log entries were generated after the specified **fromDate**. I have added a [test](https://github.com/cnagadya/openmrs-module-idgen/blob/aae65f4f52c23a0acfee80a0e0f524ffd3067502/omod/src/test/java/org/openmrs/module/idgen/rest/search/LogEntrySearchHandlerTest.java#L161-L169) for invalid date

Reply:
Since you are throwing an exception, the method name should change to reflect that.

Reply:
Since you need to test to ensure that the rest client gets an error message which gives them a clue about what the invalid field is, can you use this? https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/api/OrderServiceTest.java#L149-L150

Reply:
Thanks for sharing the resource. I have made the recommended adjustments to the test